server/graphql/loaders/expenses.ts

```diff
@@ -81,23 +81,13 @@ export const attachedFiles = (): DataLoader<number, ExpenseAttachedFile[]> => {
 /**
  * Expense loader to check if userTaxForm is required before expense payment
  */
-export const userTaxFormRequiredBeforePayment = (): DataLoader<number, boolean> => {
+export const taxFormRequiredBeforePayment = (): DataLoader<number, boolean> => {
   return new DataLoader<number, boolean>(async (expenseIds: number[]): Promise<boolean[]> => {
     const expenseIdsPendingTaxForm = await queries.getTaxFormsRequiredForExpenses(expenseIds);
     return expenseIds.map(id => expenseIdsPendingTaxForm.has(id));
   });
 };
 
-/**
- * Loader for expense's requiredLegalDocuments.
- */
-export const requiredLegalDocuments = (): DataLoader<number, string[]> => {
-  return new DataLoader(async (expenseIds: number[]) => {
-    const expenseIdsPendingTaxForm = await queries.getTaxFormsRequiredForExpenses(expenseIds);
-    return expenseIds.map(id => (expenseIdsPendingTaxForm.has(id) ? [LEGAL_DOCUMENT_TYPE.US_TAX_FORM] : []));
-  });
-};
-
```

Comment:
Both loaders were pretty much doing the same thing, I think it was a leftover from GraphQL v1. I unified them to leverage the full power of dataloaders caching.

Reply:
Good call.